bentoml/frameworks/pytorch.py

```diff
@@ -141,7 +141,10 @@ def set_dependencies(self, env: BentoServiceEnv):
             "BentoService definition file or manually add them via "
             "`@env(pip_packages=['torchvision'])` when defining a BentoService"
         )
-        env.add_pip_packages(['torch'])
+        # Hmm maybe create a function to check if our options for infer_pip_package is True?
+        # TODO: differentiate when to add torch or torch-gpu wheels
+        if env._infer_pip_packages:
```

Comment:
This is a hacky solutions to prevent env from adding torch to requirements.txt

Reply:
@aarnphm could we actually add this for all the other model artifacts as well?  I think it's an important issue that we try to add additional pip packages to the list when user specifically used "infer_pip_packages=False".

Reply:
Hmm I can add this to every artifacts if we want to allow access to protected fields like this

Reply:
@aarnphm let's add a proper `property getter` for `infer_pip_package`?

Reply:
Feel free to do that in a follow-up PR